Observer/CancelOrderItemObserver.php

```diff
@@ -131,50 +55,9 @@ public function execute(EventObserver $observer): void
             return;
         }
 
-        $websiteId = $orderItem->getStore()->getWebsiteId();
-        $websiteCode = $this->websiteRepository->getById($websiteId)->getCode();
-        $salesChannel = $this->salesChannelFactory->create([
-            'data' => [
-                'type' => SalesChannelInterface::TYPE_WEBSITE,
-                'code' => $websiteCode
-            ]
-        ]);
-
-        $salesEvent = $this->salesEventFactory->create([
-            'type' => SalesEventInterface::EVENT_ORDER_CANCELED,
-            'objectType' => SalesEventInterface::OBJECT_TYPE_ORDER,
-            'objectId' => (string)$orderItem->getOrderId()
-        ]);
-
-        //This does nothing, is disabled on the PlaceReservationsForSalesEventPlugin
-        $this->placeReservationsForSalesEvent->execute($itemsToCancel, $salesChannel, $salesEvent);
-
+        /** @var OrderInterface $order */
         $order = $orderItem->getOrder();
 
-        // Source selection happens by default on shipment, so only shipments contain information about the
-        // inventory source.
-        // @TODO add support for multi source stock replenishment on cancellation
-        if ($this->isSingleSourceMode->execute()) {
-            $sourceCode = $this->defaultSourceProvider->getCode();
-
-            $itemsToDeduct = [];
-            foreach ($itemsToCancel as $itemToCancel) {
-                $itemsToDeduct[] = $this->itemToDeductFactory->create([
-                    'sku' => $itemToCancel->getSku(),
-                    'qty' => -$itemToCancel->getQuantity(),
-                ]);
-            }
-
-            $sourceDeductionRequest = $this->sourceDeductionRequestFactory->create([
-                'sourceCode' => $sourceCode,
-                'items' => $itemsToDeduct,
-                'salesChannel' => $salesChannel,
-                'salesEvent' => $salesEvent
-            ]);
-
-            $this->sourceDeductionService->execute($sourceDeductionRequest);
-
-            $this->priceIndexer->reindexRow($orderItem->getProductId());
-        }
+        $this->executeSourceDeductionForItems->executeSourceDeductionForItems($order, $itemsToCancel);
```

Comment:
Again, we can just pass `$orderItem->getOrder()` to this functionality rather than define the order as variable first